picard/ui/searchdialog.py

```diff
@@ -0,0 +1,301 @@
+        self.buttonBox.accepted.connect(self.accept)
+        self.buttonBox.rejected.connect(self.reject)
+        self.verticalLayout.addWidget(self.buttonBox)
+
+    def add_widget_to_center_layout(self, widget):
+        wid = self.center_layout.itemAt(0)
+        if wid:
+            wid.widget().deleteLater()
+        self.center_layout.addWidget(widget)
+
+    def show_progress(self):
+        self.progress_widget = QtGui.QWidget(self)
+        layout = QtGui.QVBoxLayout(self.progress_widget)
+        text_label = QtGui.QLabel('<strong>Fetching results...</strong>', self.progress_widget)
+        text_label.setAlignment(QtCore.Qt.AlignHCenter | QtCore.Qt.AlignBottom)
+        gif_label = QtGui.QLabel(self.progress_widget)
+        movie = QtGui.QMovie(":/images/loader.gif")
+        gif_label.setMovie(movie)
+        movie.start()
+        gif_label.setAlignment(QtCore.Qt.AlignHCenter | QtCore.Qt.AlignTop)
+        layout.addWidget(text_label)
+        layout.addWidget(gif_label)
+        layout.setMargin(1)
+        self.progress_widget.setLayout(layout)
+        self.add_widget_to_center_layout(self.progress_widget)
+
+    def show_table(self):
+        self.tracksTable = TracksTable()
+        self.tracksTable.cellDoubleClicked.connect(self.track_double_clicked)
+        self.restore_table_header_state()
+        self.add_widget_to_center_layout(self.tracksTable)
+
+    def show_error(self, error):
+        self.error_widget = QtGui.QLabel(_("<strong>" + error + "</strong>"))
+        self.error_widget.setAlignment(QtCore.Qt.AlignCenter)
+        self.error_widget.setWordWrap(True)
+        self.add_widget_to_center_layout(self.error_widget)
+
+    def load_selection(self, row=None):
+        track_id, release_id, rg_id = self.search_results[row][:3]
+        if release_id:
+            self.tagger.get_release_group_by_id(rg_id).loaded_albums.add(
+                    release_id)
+            if self.file_:
+                album = self.file_.parent.album
+                self.tagger.move_file_to_track(self.file_, release_id, track_id)
+                if album._files == 0:
+                    # Remove album if the selected file was the only one in album
+                    # Compared to 0 because file has already moved to another album
+                    # by move_file_to_track
+                    self.tagger.remove_album(album)
+            else:
+                self.tagger.load_album(release_id)
+
+    def track_double_clicked(self, row):
+        self.load_selection(row)
+        self.accept()
+
+    def accept(self):
+        try:
+            sel_rows = self.tracksTable.selectionModel().selectedRows()
+            if sel_rows:
+                sel_row = sel_rows[0].row()
+                self.load_selection(sel_row)
+            self.save_state(True)
+            QtGui.QDialog.accept(self)
+        except AttributeError:
+            self.save_state(False)
+            QtGui.QDialog.accept(self)
```

Comment:
`QtGui.QDialog.accept(self)` is at the end of both try and except blocks, any particular reason to not move it outside